### PR TITLE
Issue 1460  pytest params

### PR DESCRIPTION
**Pull request by kohr-h**

### CODE REVIEW

**Comment by kohr-h** [comment]
I'll fix the PEP8 issues.

File: odl/test/largescale/tomo/analytic_slow_test.py [hunk 1]
```diff
@@ -40,14 +40,22 @@
...
+projector_ids = [
+    " geom='{}' - impl='{}' - angles='{}' ".format(*p.values[0].split())
+    for p in projectors
+]
 
 # bug in pytest (ignores pytestmark) forces us to do this this
 largescale = " or not pytest.config.getoption('--largescale')"
-projectors = [pytest.mark.skipif(p.args[0] + largescale, p.args[1])
-              for p in projectors]
+projectors = [
```
Comment (adler-j) [inline odl/test/largescale/tomo/analytic_slow_test.py]: Is this truly needed? It's honestly hard to read.

File: odl/test/largescale/tomo/ray_transform_slow_test.py [hunk 2]
```diff
@@ -49,14 +48,23 @@
...
+    " geom='{}' - impl='{}' - angles='{}' ".format(*p.values[0].split())
+    for p in projectors
+]
 
 
 # bug in pytest (ignores pytestmark) forces us to do this this
 largescale = " or not pytest.config.getoption('--largescale')"
-projectors = [pytest.mark.skipif(p.args[0] + largescale, p.args[1])
-              for p in projectors]
+projectors = [
```
Comment (adler-j) [inline odl/test/largescale/tomo/ray_transform_slow_test.py]: Same as above here

File: odl/test/largescale/tomo/analytic_slow_test.py (hunk 1 above)
Comment (kohr-h) [reply odl/test/largescale/tomo/analytic_slow_test.py]: It ain't pretty, I agree. Maybe I'll just make a little function to do the dirty work.

Formatting-wise, I'm currently going with what the [`black`](https://github.com/ambv/black) tool spits out. It usually does a good job and makes code formatting a no-brainer. Sometimes it looks a bit like Lisp. Is that a bad thing? :stuck_out_tongue:

Comment (adler-j) [reply odl/test/largescale/tomo/analytic_slow_test.py]: Lets go ahead for now, but long term I'm not sure if it is a bit to verbose

Comment (kohr-h) [reply odl/test/largescale/tomo/analytic_slow_test.py]: Actually that code didn't work at all, as I found out. I changed the implementation, now I think it's much cleaner. Please check again.

File: odl/test/largescale/tomo/ray_transform_slow_test.py [hunk 3]
```diff
@@ -11,52 +11,61 @@
...
-
-
-# bug in pytest (ignores pytestmark) forces us to do this this
-largescale = " or not pytest.config.getoption('--largescale')"
-projectors = [pytest.mark.skipif(p.args[0] + largescale, p.args[1])
-              for p in projectors]
+projectors = []
+projectors.extend(
+    (pytest.param(value, marks=skip_if_no_astra)
+     for value in ['par2d astra_cpu uniform',
```
Comment (adler-j) [inline odl/test/largescale/tomo/ray_transform_slow_test.py]: Prefer something other than "value", e.g. be specific.

Comment (kohr-h) [reply odl/test/largescale/tomo/ray_transform_slow_test.py]: Well, `values` is the name under which the input is stored in `pytest.param`. But it's a generic construct, so you're right, we can be somewhat more specific here.

**Pull request merged**